client/src/components/_common/AppIcon/AppIcon.jsx

```diff
@@ -4,10 +4,14 @@ import PropTypes from 'prop-types';
 import Icon from '_common/Icon';
 import './AppIcon.scss';
 
-const AppIcon = ({ appId }) => {
+const AppIcon = ({ appId, category }) => {
   const appIcons = useSelector((state) => state.apps.appIcons);
   const findAppIcon = (id) => {
-    let appIcon = 'applications';
+    if (!category) {
+      console.error('Category is undefined for appId:', id);
+      return 'applications';
```

Comment:
**I think this is a good error to have. 👍**

The return string is interesting. What is the result of "applications" compared to nothing? (I tested it by changing the condition to true, and visiting apps, but I did not see "applications" in the console.)

Reply:
My theory was this conditional would help if there were other parts of the code where the <AppIcon> component was being rendered and no category prop was present/passed. For example, initially in AppForms I was getting an 'undefined' for the category related to the icon with the documentation link. Until I added that new function getAppCategory. But I think in this case I can out this conditional bc I didn't make the category prop in AppIcon required--and 2) I gave it a default value of 'applications' anyway, so in theory it should always: 1) have a custom icon 2) default to whatever category icon it's associated with 3) if all else fails default to the applications category for the icon.

Reply:
Took this conditional out. And wrote a function that defaults the icon to 'icon-applications' if the category's className doesn't exist in the stylesheets.